tests/test_TFEngine.py

```diff
@@ -1078,6 +1078,86 @@ def test_attention_train_then_search():
   engine.finalize()
 
 
+def test_attention_no_encoder_dependency():
+  from GeneratingDataset import DummyDataset
+  seq_len = 5
+  n_data_dim = 2
+  n_classes_dim = 7
+  train_data = DummyDataset(input_dim=n_data_dim, output_dim=n_classes_dim, num_seqs=2, seq_len=seq_len)
+  train_data.init_seq_order(epoch=1)
+  dev_data = DummyDataset(input_dim=n_data_dim, output_dim=n_classes_dim, num_seqs=2, seq_len=seq_len)
+  dev_data.init_seq_order(epoch=1)
+
+  from TFUtil import DimensionTag
+  enc_time = DimensionTag(kind=DimensionTag.Types.Spatial, description="enc time")
+
+  config = Config()
```

Comment:
You should include `debug_print_layer_output_template` (and maybe other things) in the config. Like:
```suggestion
  config = Config({"debug_print_layer_output_template": True})
```

Reply:
Ah sorry, did not see that you call `update` then. (You could skip that by a single call...)